src/main/java/cd/go/artifact/docker/registry/model/ArtifactPlanConfigTypeAdapter.java

```diff
@@ -39,10 +39,22 @@ public JsonElement serialize(ArtifactPlanConfig src, Type typeOfSrc, JsonSeriali
     }
 
     private boolean isBuildFileConfig(JsonObject jsonObject) {
-        return jsonObject.has("BuildFile") && !jsonObject.has("Image") && !jsonObject.has("Tag");
+        return containsBuildFileProperty(jsonObject) && !containsImageProperty(jsonObject);
     }
 
     private boolean isImageTagConfig(JsonObject jsonObject) {
-        return jsonObject.has("Image") && !jsonObject.has("BuildFile");
+        return containsImageProperty(jsonObject) && !containsBuildFileProperty(jsonObject);
+    }
+
+    private boolean containsBuildFileProperty(JsonObject jsonObject) {
+        return jsonObject.has("BuildFile") && isPropertyNotBlank(jsonObject, "BuildFile");
+    }
+
+    private boolean containsImageProperty(JsonObject jsonObject) {
+        return jsonObject.has("Image") && isPropertyNotBlank(jsonObject, "Image");
+    }
+
+    private boolean isPropertyNotBlank(JsonObject jsonObject, String tag) {
```

Comment:
Calling this parameter `tag` may be a bit confusing. Maybe call it `property` instead?